python/ray/train/lightning/lightning_checkpoint.py

```diff
@@ -0,0 +1,61 @@
+from typing import TYPE_CHECKING, Optional, Type, Union, Dict
+
+from ray.air.checkpoint import Checkpoint
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    from pytorch_lightning import LightningModule
+    from ray.data.preprocessor import Preprocessor
+
+
+@PublicAPI(stability="alpha")
+class LightningCheckpoint(Checkpoint):
```

Comment:
@sumanthratna I believe we would need an accompanying LightningPredictor with this Checkpoint to do inference like so https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#inference

Reply:
Oh, should I add LightningPredictor to this PR as well? I think I asked @matthewdeng in Slack and I think he said we can merge LightningPredictor after LightningTrainer

Reply:
ok sg, lets do that in a follow up